pkger/parser_test.go

```diff
@@ -4825,6 +4829,249 @@ func Test_validGeometry(t *testing.T) {
+			expErr: "",
+		},
+		// invalid
+		{
+			path:   "i\nvalid:///foo",
+			extra:  false,
+			expErr: "invalid filepath provided",
+		},
+		{
+			path:   "testdata/nonexistent (darwin|linux)",
+			extra:  false,
+			expErr: "no such file or directory",
+			oses:   []string{"darwin", "linux"},
+		},
+		{
+			path:   "testdata/nonexistent (windows)",
+			extra:  false,
+			expErr: "The system cannot find the file specified.",
+			oses:   []string{"windows"},
+		},
+		// invalid with extra
+		{
+			path:   "/dev/null",
+			extra:  true,
+			expErr: "file in special file system",
+			oses:   []string{"linux"},
+		},
+		// symlink to /dev/null, invalid with extra
+		{
+			path:   devNullSym,
+			extra:  true,
+			expErr: "file in special file system",
+			oses:   []string{"linux"},
+		},
+		// invalid with extra
+		{
+			path:   "/dev/null",
+			extra:  true,
+			expErr: "not a regular file",
+			oses:   []string{"darwin"},
+		},
+		{
+			path:   "/",
+			extra:  true,
+			expErr: "not a regular file",
+		},
+		{
+			path:   "testdata/nonexistent (darwin|linux)",
+			extra:  true,
+			expErr: "no such file or directory",
+			oses:   []string{"darwin", "linux"},
+		},
+		{
+			path:   "testdata/nonexistent (windows)",
+			extra:  true,
+			expErr: "The system cannot find the file specified.",
+			oses:   []string{"windows"},
+		},
+		{
+			path:   emptyFn,
+			extra:  true,
+			expErr: "file empty",
+		},
+		{
+			path:   bigFn,
+			extra:  true,
+			expErr: "file too big",
+		},
+	}
+
+	// Add tmpfs special file system exception tests for Linux.
+	// We don't consider errors creating the tests cases (e.g. no tmpfs mounts, no write
+	// permissions to any tmpfs mount) errors for the test itself. It's possible we
+	// can't run these tests in some locked down environments, but we will warn the
+	// use we couldn't run the tests.
+	if runtime.GOOS == "linux" {
+		tmpfsDirs, err := func() ([]string, error) {
+			t.Helper()
+			// Quick and dirty parse of  /proc/mounts to find tmpfs mount points on system.
+			mounts, err := os.ReadFile("/proc/mounts")
+			if err != nil {
+				return nil, fmt.Errorf("error reading /proc/mounts: %w", err)
+			}
+			mountsLines := strings.Split(string(mounts), "\n")
+			var tmpfsMounts []string
+			for _, line := range mountsLines {
+				if line == "" {
+					continue
+				}
+				cols := strings.Split(line, " ")
+				if len(cols) != 6 {
+					return nil, fmt.Errorf("unexpected /proc/mounts line format (%d columns): %q", len(cols), line)
+				}
+				if cols[0] == "tmpfs" {
+					tmpfsMounts = append(tmpfsMounts, cols[1])
+				}
+			}
+			if len(tmpfsMounts) == 0 {
+				return nil, errors.New("no tmpfs mount points found")
+			}
+
+			// Find which common tmpfs directories are actually mounted as tmpfs.
+			candidateDirs := []string{
+				"/dev/shm",
+				"/run",
+				"/tmp",
+				os.Getenv("XDG_RUNTIME_DIR"),
+				os.TempDir(),
+			}
+			var actualDirs []string
+			for _, dir := range candidateDirs {
+				if dir == "" {
```

Comment:
Nitpick: either here or the `os.Getenv("XDG_RUNTIME_DIR")` might benefit from a comment that this is for the other. Not a blocker.